src/shogun/mathematics/graph/ops/shogun/details/load_avx.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include "PacketType.h"
+#include <Eigen/Core>
+
+using namespace Eigen::internal;
+
+namespace shogun::graph::op {
+			
+	template <typename T>
+	aligned_vector load_avx(void* input1);
+
+	template <typename T>
+	aligned_vector load_avx(void* data)
```

Comment:
the only detail missing is here. I am not 100% sure that msvc will allow us to send data back and forth using variant... At worst I can cast it to void* as I did with all the rest